sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java

```diff
@@ -72,6 +79,38 @@ public HttpLogOptions setLogLevel(final HttpLogDetailLevel logLevel) {
         return this;
     }
 
+    /**
+     * Gets the application specific id.
+     *
+     * @return The application specific id.
+     */
+    public String getApplicationId() {
+        return applicationId;
+    }
+
+    /**
+     * Sets the custom application specific id supplied by the user of the client library.
+     *
+     * @param applicationId The user specified application id.
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setApplicationId(final String applicationId) {
+        if (applicationId != null
```

Comment:
I think the logic here could be simplified, right now this is going to perform the same check multiple times.

``` java
if (!CoreUtils.isNullOrEmpty(applicationId)) {  // Would an issue arise if the application ID is an empty string? Should we disallow that?
    if (applicationId.contains(" ")) {
        // throw error
    } else if (applicationId.length() > 24) {
        // throw error
    } else {
        this,applicationId = applicationId;
    }
}

return this;
```

Reply:
Simplified the logic.
Currently, we are allowing empty application Id.
@JonathanGiles do we want to disallow empty application Id?